lib/pbench/test/unit/agent/test_tool_data_sink.py

```diff
@@ -31,12 +31,16 @@ def cleanup_tmp(self, tmp_path):
         yield
         try:
             shutil.rmtree(self.int_pb_run)
-        except Exception as exc:
-            print(exc)
+        except FileNotFoundError:
+            pass
+        except NotADirectoryError:
+            self.int_pb_run.unlink()
```

Comment:
Hmm; this is cleanup code after the `yield`, so here we're *accepting* a test run that somehow creates a file where we expect either nothing (is it actually OK if the associated test cases don't even create the directory?) or a directory. Doesn't it sound like an error in the tests that we might detect here?

And the previous code did the `print(exc)` (which is rather anemic) if the directory didn't exist ... should it actually be an error if the test case didn't create the directory?

Reply:
What do you think this should be?

I don't know of another way to do this, other than have the individual tests not use a fixture, but create and destroy what is appropriate for their test case all within the test.

We can use `shutil.rmtree()` on a non-existent file, and the behavior of the test determines whether or not it gets created.

We can use `Path.unlink()` because it might be a directory, and the directory hierarchy won't get removed.

So this seemed like the best approach.

If you have another way to do this without re-writing unit tests (which would be a whole 'nother level of effort) I am all ears.

Reply:
First, I think that this change is better than it was.

Beyond that, I agree that it would be better still if this code detected incorrect behavior and caused the test to fail.

However, in this context (without adding a little superstructure), I don't think we can tell whether the file/directory should have existed or not.  (That is, we might have tests which use this fixture which, in their "success scenario", don't create both files...in which case we don't want this fixture to generate a failure for that.)

Reply:
So can we resolve this?

Reply:
I'm good with it the way it is currently proposed to be.

Reply:
Yeah; if we have tests that really do create regular files here, then Webb's correct that the mechanism to manage failure here probably isn't justified. Ah, well.